frontend/src/components/transactions/Transactions.tsx

```diff
@@ -15,47 +17,57 @@ export interface State {
   transactions: T.Transaction[] | null;
 }
 
-export default class extends React.PureComponent<Props, State> {
+export default class extends React.Component<Props, State> {
   static defaultProps = {
     reversed: false,
-    limit: 50
+    limit: 15
   };
 
   state: State = {
     transactions: null
   };
 
   _transactionsApi: TransactionsApi | null;
+  timer: any;
 
   constructor(props: Props) {
     super(props);
 
     // TODO: Design ExplorerApi to handle server-side rendering gracefully.
     this._transactionsApi = null;
+    this.timer = null;
   }
 
   componentDidMount() {
-    this.fetchTransactions();
+    this.timer = setTimeout(this.regularFetchInfo, 0);
+    this.regularFetchInfo();
   }
 
-  componentDidUpdate(prevProps: Props) {
-    if (this.props !== prevProps) {
-      this.fetchTransactions();
-    }
+  componentWillUnmount() {
+    clearTimeout(this.timer);
+    this.timer = null;
   }
 
-  render() {
-    const { transactions } = this.state;
-    if (transactions === null) {
-      return null;
+  regularFetchInfo = async () => {
+    if (this.state.transactions === null) {
+      await this.fetchTransactions();
+    } else {
+      await this.fetchInfo();
```

Comment:
The naming is confusing, and it does not seem to be right to fetch the latest transactions, as this component is widely used in other places (Block details, Account details, etc, where you want the latest transactions **filtered** by the specific condition). You should be fine with just calling `this.fetchTransactions()` all the time. `fetchInfo` seems redundant to me.